### PR TITLE
 #1915 Should separate collie creation to two stages also

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1915)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/com/island/ohara/agent/Collie.scala [hunk 1]
```diff
@@ -135,6 +135,13 @@ trait Collie[T <: ClusterInfo, Creator <: ClusterCreator[T]] {
     * @return true if it does remove a node from a running cluster. Otherwise, false
     */
   def removeNode(clusterName: String, nodeName: String)(implicit executionContext: ExecutionContext): Future[Boolean]
+
+  // API will need two stages to start a cluster (create -> start),
+  // but collie does not need to store "cluster information" for creation.
+  // We just add a "return cluster" function for testing only
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/Collie.scala]: we should NOT add code which is used in testing only to interface.

perhaps we can replace the impl of zk_start by "cluster(clusterName).map(_._1)"

Comment (saivirtue) [reply ohara-agent/src/main/scala/com/island/ohara/agent/Collie.scala]: Sure.

**Comment by saivirtue** [comment]
retry it

**Comment by saivirtue** [comment]
retry it

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
retry it

**Pull request merged**